scripts/west_commands/run_common.py

```diff
@@ -284,13 +284,14 @@ def do_run_common(command, user_args, user_runner_args, domain_file=None):
         if len(entry.boards) == 0:
             del used_cmds[i]
 
+    prev_runner = None
     for d in domains:
-        do_run_common_image(command, user_args, user_runner_args,
-                            used_cmds, board_image_count, d.build_dir)
+        prev_runner = do_run_common_image(command, user_args, user_runner_args, used_cmds,
+                                          board_image_count, prev_runner, d.build_dir)
 
 
 def do_run_common_image(command, user_args, user_runner_args, used_cmds,
-                        board_image_count, build_dir=None,):
+                        board_image_count, prev_runner, build_dir=None, ):
```

Comment:
Wouldn't it be best to add new arguments at the end? This function probably isn't called from OOT users, but better safe than sorry?

```suggestion
                        board_image_count, build_dir=None, prev_runner=None, ):
```

Reply:
I didn't add it to the end because `build_dir` has the default value (not sure why TBH, as the call always provides the value). I can swap the ordering, but I don't think it really makes a difference.

Reply:
It will make a difference if OOT calls of `do_run_comm_image` have passed a build directory using positional arguments. Suddenly your `prev_runner` is passed the build directory and `build_dir` will get the default `None` value.

Reply:
Sure, but AFAICS the function **can** be used from any command.

I'm not saying this an issue, it's simply defensive programming to avoid edge-cases and annoyances.

Reply:
Is this all that is stopping an approval? If so I will change it

Reply:
I don't currently block it, but it's an easy fix IMO, and I would approve if you change it.